homeassistant/components/rki_covid/__init__.py

```diff
@@ -0,0 +1,147 @@
+"""RKI Covid numbers integration."""
+import asyncio
+from datetime import timedelta
+import logging
+
+import aiohttp
+import async_timeout
+from rki_covid_parser.parser import RkiCovidParser
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import aiohttp_client, update_coordinator
+
+from .const import DOMAIN
+from .data import DistrictData, StateData
+
+_LOGGER = logging.getLogger(__name__)
+PLATFORMS = ["sensor"]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up component from a config entry."""
+    _LOGGER.debug(f"Setup item from config entry: {entry.data}.")
+    # Forward the setup to the sensor platform.
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Unload a config entry."""
+    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+
+async def get_coordinator(hass: HomeAssistant, parser: RkiCovidParser):
+    """Get the data update coordinator."""
+    _LOGGER.debug("initialize the data coordinator.")
+    if DOMAIN in hass.data:
+        return hass.data[DOMAIN]
+
+    async def async_get_districts():
+        """Fetch data from rki-covid-parser library.
+
+        Here the data for each district is loaded.
+        """
+        _LOGGER.debug("fetch data from rki-covid-parser.")
+        try:
+            with async_timeout.timeout(30):
+                # return {case.county: case for case in await api.load_districts()}
+                await parser.load_data()
+                _LOGGER.debug("fetching finished.")
+
+                items = {}
+
+                # districts
+                for d in parser.districts:
+                    district = parser.districts[d]
+
+                    items[district.county] = DistrictData(
```

Comment:
It seems that you are the author of `rki_covid_parser`. Why don't you directly add this typing in your python library?

Reply:
That is correct, the parser is only for downloading and parsing the data.
Since merging is a frontend requirement, it should not be part of a parsing library